source/numeric-range.d.ts

```diff
@@ -0,0 +1,46 @@
+/**
+A `number` that is an integer.
+
+Use-case: Validating and documenting parameters.
+
+@example
+```
+declare function setYear<T extends number>(length: Integer<T>): void;
+```
+*/
+export type Integer<T extends number> = `${T}` extends `${bigint}` ? T : never;
+
+type Numeric = number | bigint;
+
+type Zero = 0 | 0n;
+
+/**
+A negative `number`/`bigint` (`(-∞, 0)`).
+
+Use-case: Validating and documenting parameters.
+*/
+export type Negative<T extends Numeric> = T extends Zero ? never : `${T}` extends `-${string}` ? T : never;
+
+/**
+A positive `number`/`bigint` (`(0, ∞)`).
+
+Use-case: Validating and documenting parameters.
+
+@example
+```
+declare function setQuantity<T extends number>(length: Positive<T>): void;
+```
+*/
+export type Positive<T extends Numeric> = T extends Zero ? never : Negative<T> extends never ? T : never;
+
+/**
+An natural `number`/`bigint` (`[0, ∞)`).
```

Comment:
I would explain for explicitly what a natural number is.

Reply:
"`[0, ∞)`" is supposed to describe it as an [interval](https://en.wikipedia.org/wiki/Interval_(mathematics)).
I could replace that or add some text of "positive numbers and 0".

Reply:
I think it's fine to have it like that, but add another sentence below with:

> A natural number is a positive number or zero.